static/js/poll_modal.js

```diff
@@ -0,0 +1,66 @@
+import $ from "jquery";
+import {Sortable} from "sortablejs";
+
+import render_poll_modal_option from "../templates/poll_modal_option.hbs";
+
+export function update_option_delete_btn($container, display_flag) {
+    const no_of_option_row = $container.find(".option-row").length;
+
+    // Disable delete button if there only one choice row
+    // Enable choice delete button more one than once choice
+    if (no_of_option_row === 1) {
+        if (display_flag === true) {
+            $container.find(".option-row .delete-option").show();
+        } else {
+            $container.find(".option-row .delete-option").hide();
+        }
+    }
+}
+
+function create_option_row(container) {
+    const row = render_poll_modal_option();
+    $(container).append(row);
+}
+
+function add_option_row(e) {
+    if ($(e.target).parent().next().hasClass("option-row")) {
+        return;
+    }
+    const options_div = e.delegateTarget;
+    update_option_delete_btn($(options_div), true);
+    create_option_row(options_div);
+}
+
+function delete_option_row(e) {
+    const $row = $(e.currentTarget).parent();
+    const $container = $row.parent();
+    $row.remove();
+    update_option_delete_btn($container, false);
+}
+
+export function poll_options_setup() {
+    const $poll_options_list = $("#add-poll-form .poll-options-list");
+
+    create_option_row($poll_options_list);
+    update_option_delete_btn($poll_options_list, false);
+
+    Sortable.create($poll_options_list[0], {
+        onUpdate() {},
+        filter: "input",
+        preventOnFilter: false,
+    });
+
+    $poll_options_list.on("input", ".option-row input", add_option_row);
+    $poll_options_list.on("click", "button.delete-option", delete_option_row);
+}
+
+export function frame_poll_message_content() {
```

Comment:
Your existing test could do well by adding a few more mock poll options and then finally calling this function to assert its output.

Reply:
Not feasible due to limits and unsuitability of zjquery for this case, so current tests will have to do